tests/test_types.py

```diff
@@ -699,6 +714,22 @@ def test_constrained_str_too_long():
     ]
 
 
+def test_constrained_str_upper_enabled():
+    class Model(BaseModel):
+        v: constr(to_upper=True)
+
+    m = Model(v='abcd')
+    assert m.v == 'ABCD'
+
+
+def test_constrained_str_upper_disabled():
+    class Model(BaseModel):
+        v: constr(to_upper=False)
+
+    m = Model(v='aBcD')
+    assert m.v == 'aBcD'
+
+
```

Comment:
I think we can use `pytest.mark.parametrize` for these two tests here as well

Reply:
refactored on ccb42d3